daml/splice-dso-governance/daml/Splice/DsoRules.daml

```diff
@@ -1532,6 +1558,45 @@ template DsoRules with
           exercise unclaimedActivityRecordCid UnclaimedActivityRecord_DsoExpire
         pure DsoRules_ExpireUnclaimedActivityRecordResult with unclaimedRewardCid
 
+    nonconsuming choice DsoRules_ValidatorLicense : DsoRules_ValidatorLicenseResult
+      with
+        actions : [ValidatorLicenseActionKind]
+      controller dso
+      do now <- getTime
+         let expiresAt = now `addRelTime` getVoteExecutionInstructionTimeout config
+         instructionCids <- forA actions $ \action -> do
+           case action of
+             VLAK_ChangeWeight {validator, weight} -> do
+               require ("Weight is non-negative for " <> show validator) (weight >= 0.0)
+               let instruction = EI_ValidatorLicenseChangeWeight with validator; weight
+               create VoteExecutionInstruction with dso; instruction; expiresAt
+             VLAK_Withdraw {validator} -> do
+               let instruction = EI_ValidatorLicenseWithdraw with validator
+               create VoteExecutionInstruction with dso; instruction; expiresAt
+         return DsoRules_ValidatorLicenseResult with instructionCids
+
+    nonconsuming choice DsoRules_Execute_VoteInstruction : DsoRules_Execute_VoteInstructionResult
```

Comment:
```suggestion
    nonconsuming choice DsoRules_ExecuteVoteInstruction : DsoRules_Execute_VoteInstructionResult
```
my current simple rule: `_` only used to separate template from action, but action is in CamelCase.